services/viva-ms/lambdas/submitApplication.js

```diff
@@ -3,86 +3,132 @@ import AWS from 'aws-sdk';
 import to from 'await-to-js';
 
 import config from '../../../config';
+
+import * as dynamoDb from '../../../libs/dynamoDb';
+import log from '../../../libs/logs';
 import params from '../../../libs/params';
-import { putEvent } from '../../../libs/awsEventBridge';
+
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
 import vivaAdapter from '../helpers/vivaAdapterRequestClient';
-import log from '../../../libs/logs';
 
 const VIVA_CASE_SSM_PARAMS = params.read(config.cases.providers.viva.envsKeyName);
+const VIVA_APPLICATION_RECEIVED = 'VIVA_APPLICATION_RECEIVED';
 
 const dynamoDbConverter = AWS.DynamoDB.Converter;
 
 export async function main(event, context) {
-  if (event.detail.dynamodb.NewImage === undefined) {
+  if (event.detail.dynamodb.NewImage == undefined) {
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if(noSuchProperty('dynamodb', event.detail)) {
   // do something
 }

```

Reply:
Perhaps:

```js
if (!event.detail?.dynamodb?.NewImage) {
...
}
```

Reply:
Will be fixed in future task